### ISSUE TITLE
make eosio.token inheritable

### ISSUE THREAD

**Issue by cc32d9** [EOSIO/eosio.token#12]
It should be possible to inherit from eosio.token class, so that new tokens would not have to just copy the token source code.

So, all flesh should be moved to a token.hpp inside eosiolib, and eosio.token.cpp should only have EOSIO_ABI call.

**Comment by deckb** [comment]
Transferring this issue to the new reference token contract repo.

**Comment by mschoenebeck** [comment]
Agree. The token contract should be inheritable. Any progress on this? I would like to extend the base functionality for shielded transactions using zk-snarks and make it available for everyone. Inheritance of the base eosio.token contract would be the most elegant solution to this.
